conversion/conversion.go

```diff
@@ -242,6 +242,39 @@ func performSnapshotMigration(config writer.BatchWriterConfig, conv *internal.Co
 	return batchWriter
 }
 
+func performDataprocMigration(sourceProfile profiles.SourceProfile, targetProfile profiles.TargetProfile, config writer.BatchWriterConfig, conv *internal.Conv, client *sp.Client, infoSchema common.InfoSchema) (*writer.BatchWriter, error) {
+	common.SetRowStats(conv, infoSchema)
+	batchWriter := populateDataConv(conv, config, client)
+
+	//Adding all dataproc configs to a map
+	dataprocConfig := map[string]string{}
+	dataprocConfig["hostname"] = sourceProfile.Conn.Mysql.Host
+	dataprocConfig["port"] = sourceProfile.Conn.Mysql.Port
+	dataprocConfig["user"] = sourceProfile.Conn.Mysql.User
+	dataprocConfig["pwd"] = sourceProfile.Conn.Mysql.Pwd
+	dataprocConfig["targetdb"] = targetProfile.Dc.TargetDB
+	dataprocConfig["instance"] = targetProfile.Conn.Sp.Instance
+	dataprocConfig["project"] = targetProfile.Conn.Sp.Project
```

Comment:
What is the advantage of merging `sourceProfile` and `targetProfile` objects into a single map?
Can we carry source and target values in their own respective objects inside `ProcessDataWithDataproc` method? This would ensure better readability IMO

Reply:
Moved all dataproc api request handling to a new `dproc` package
Commit: https://github.com/cloudspannerecosystem/harbourbridge/pull/520/commits/c93cd29b12cb4aa416ae6d4d5d8e90c629e017df